server/src/core/database-service.ts

```diff
@@ -207,126 +207,135 @@ export class DatabaseService {
               }
             }
           ]
         };
         
         try {
-          const results = await this.qdrantClient.scroll(COLLECTION_NAME, {
-            filter: filter,
-            limit: 1,
-            with_payload: true
-          });
-          
-          return results.points.length > 0;
+          const countResponse = await this.qdrantClient.count(COLLECTION_NAME, { filter });
+          return countResponse.count > 0;
         } catch (error) {
-          // If text matching is not supported, try alternative approach
-          console.error('Error searching by ID pattern:', error);
-          
-          // Alternative: Get all documents and check client-side
-          const allResults = await this.qdrantClient.scroll(COLLECTION_NAME, {
-            limit: 100, // Reasonable limit to check
-            with_payload: true
-          });
-          
-          // Check if any document ID starts with the pattern
-          return allResults.points.some(point => {
-            const payload = point.payload as any;
-            return payload.id && payload.id.startsWith(idPattern);
-          });
+          console.error('Error checking transcript existence:', error);
+          return false;
         }
       },
       this.fallbackService.isFallbackActive()
     );
   }
 
-  // Delete all chunks for a specific YouTube video by videoId
+  // Delete YouTube video transcript chunks
   public async deleteYoutubeTranscriptChunks(videoId: string): Promise<number> {
-    const idPattern = `youtube_${videoId}`;
-    
     return this.fallbackService.withFallback(
       'deleteYoutubeTranscriptChunks',
-      // Fallback function for in-memory storage
+      // Fallback function
       () => {
         let deletedCount = 0;
-        // Get all keys that start with the pattern
-        const keysToDelete = Array.from(inMemoryDocuments.keys())
-          .filter(key => {
-            const doc = inMemoryDocuments.get(key)?.document;
-            return doc && doc.id.startsWith(idPattern);
-          });
-        
-        // Delete each matching document
-        keysToDelete.forEach(key => {
-          inMemoryDocuments.delete(key);
-          deletedCount++;
+        // Filter out documents from the specified YouTube video
+        Array.from(inMemoryDocuments.keys()).forEach(key => {
+          const doc = inMemoryDocuments.get(key)?.document;
+          if (doc && doc.documentName === `youtube:${videoId}`) {
+            inMemoryDocuments.delete(key);
+            deletedCount++;
+          }
         });
-        
-        console.log(`Deleted ${deletedCount} in-memory chunks for YouTube video ${videoId}`);
         return deletedCount;
       },
       // Qdrant function
       async () => {
         try {
-          // Create a filter that matches points with id starting with the pattern
           const filter = {
             must: [
               {
-                key: 'id',
+                key: 'documentName',
```

Comment:
no, why you can make a very disruption and incompatibility breaking change like this?